modules/exploits/windows/scada/delta_ia_commgr_bof.rb

```diff
@@ -0,0 +1,75 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Msf::Exploit::Remote::Tcp
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Delta Electronics Delta Industrial Automation COMMGR 1.08 Stack Buffer Overflow',
+      'Description'    => %q{
+        This module exploits a stack based buffer overflow in Delta Electronics Delta Industrial
+        Automation COMMGR 1.08. The vulnerability exists in COMMGR.exe when handling specially
+        crafted packets. This module has been tested successfully on Delta Electronics Delta
+        Industrial Automation COMMGR 1.08 over
+          Windows XP SP3,
+          Windows 7 SP1, and
+          Windows 8.1.
+      },
+      'Author'         =>
+        [
+          't4rkd3vilz',    # PoC
```

Comment:
I believe it's traditional to attribute `ZDI` for discovery in instances where the vulnerability was reported anonymously.

```ruby
         'ZDI', # Initial discovery
```

Reply:
No function codes for remote version check in Modbus TCP. See https://en.wikipedia.org/wiki/Modbus#Supported_function_codes.

Reply:
Unresolved.